ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/NTupleOmitPrefixStorageLayoutExtension.java

```diff
@@ -0,0 +1,168 @@
+    @Override
+    public String getDescription() {
+        return "This storage root extension describes an OCFL storage layout "
+        		+ "combining a pairtree-like root directory structure derived from "
+        		+ "prefix-omitted object identifiers, followed by the prefix-omitted "
+        		+ "object identifier themselves. The OCFL object identifiers are "
+        		+ "expected to contain prefixes which are removed in the mapping to "
+        		+ "directory names. The OCFL object identifier prefix is defined as "
+        		+ "all characters before and including a configurable delimiter. "
+        		+ "Where the prefix-omitted identifier length is less than "
+        		+ "tuple size * number of tuples, the remaining object id (prefix omitted) "
+        		+ "is left or right-side, zero-padded (configurable, left default), "
+        		+ "or not padded (none), and optionally reversed (default false). "
+        		+ "The object id is then divided into N n-tuple segments, and used "
+        		+ "to create nested paths under the OCFL storage root, followed by "
+        		+ "the prefix-omitted object id directory.";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void init(OcflExtensionConfig config) {
+    	Enforce.notNull(config, "configFile cannot be null");
+        
+    	if (!(config instanceof NTupleOmitPrefixStorageLayoutConfig)) {
+    		throw new OcflExtensionException(String.format("This extension only supports %s configuration. Received: %s",
+             	getExtensionConfigClass(), config));
+     	}
+
+    	NTupleOmitPrefixStorageLayoutConfig castConfig = (NTupleOmitPrefixStorageLayoutConfig) config;
+
+     	this.config = castConfig;
+    }
+
+
+    @Override
+    public Class<? extends OcflExtensionConfig> getExtensionConfigClass() {
+        return NTupleOmitPrefixStorageLayoutConfig.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String mapObjectId(String objectId) {
+        if (config == null) {
+            throw new OcflExtensionException("This extension must be initialized before it can be used.");
+        }
+        
+        if (!objectId.matches("\\A\\p{ASCII}*\\z")) {
+        	throw new OcflExtensionException(String.format("This id %s must contain only ASCII characters.", objectId));
+        }
+        //Split by delimiter and get the last part
+        String id = objectId.toLowerCase();
+        int index = id.lastIndexOf(config.getDelimiter());
+        String section = objectId;
+        String baseObjectId = "";
+
+        if (index > -1) {
+            section = objectId.substring(index + config.getDelimiter().length());
+            baseObjectId = section;
+        }
+        else {
+        	throw new OcflExtensionException(String.format( "The delimiter %s cannot be found in %s.", config.getDelimiter(), objectId));
+        }
+        
+        if (OcflConstants.EXTENSIONS_DIR.equals(section) || section.isEmpty()) {
```

Comment:
You should probably remove the `section.isEmpty()` here as you are checking it in the next conditional. Up to you if you want to do the `length()` or `isEmpty()`.